app/src/main/java/fi/aalto/legroup/achso/entities/UserPool.java

```diff
@@ -0,0 +1,55 @@
+package fi.aalto.legroup.achso.entities;
+
+import java.util.ArrayList;
+
+/**
+ * Use this class to store users memory efficiently.
+ * Maps User objects to indices that can later retrieve the actual User back.
+ */
+public class UserPool {
```

Comment:
This is probably not used outside `.entities` either, grounds for package-private? Also, if it’s not meant to be instantiated, a `private` default constructor might be useful.